### ISSUE TITLE
Fetch available artifacts versions for given repo

### ISSUE THREAD

**Issue by cpotter302** [software-engineering-project-org/vscode-plugin-automotive-app-deployment#65]
Fetch available images from the github api and implement it in the backend logic

Prerequisites: #64 
Links: #39 

Acceptance criteria:

Reviewed code is merged into main.
Sufficient testing for Unit.
